onnxscript/ir/_core.py

```diff
@@ -1027,11 +1076,67 @@ def opset_imports(self) -> dict[str, int]:
 
     @property
     def nodes(self) -> Sequence[Node]:
-        return self._nodes
+        return tuple(self._nodes)
 
-    def topologically_sorted_nodes(self) -> Sequence[Node]:
+    def __getitem__(self, index: int) -> Node:
+        return self._nodes[index]
+
+    def __len__(self) -> int:
+        return len(self._nodes)
+
+    def __iter__(self) -> Iterator[Node]:
+        return iter(self._nodes)
+
+    def __reversed__(self) -> Iterator[Node]:
+        return reversed(self._nodes)
+
+    def _remove_graph_reference_from_node(self, node: Node) -> None:
+        """A private function to remove the graph reference from a node, used when mutating the graph by node_list."""
+        # pylint: disable=protected-access
+        node._graph = None
+        # pylint: enable=protected-access
+
+    def _add_graph_reference_to_node(self, node: Node) -> None:
+        """A private function to add the graph reference to a node, used when mutating the graph by node_list."""
+        # pylint: disable=protected-access
+        if node._graph is not None:
+            node._graph.remove(node)
+        node._graph = self
+        # pylint: enable=protected-access
+
+    # Mutation methods
+    def append(self, node: Node) -> None:
+        """Append a node to the graph in O(1) time."""
+        self._nodes.append(node, property_modifier=self._add_graph_reference_to_node)
```

Comment:
Is the property_modifier outdated? (same below.)

Reply:
Yes. Once I merge #1349 to this pr they will be gone